sympy/physics/control/lti.py

```diff
@@ -438,69 +453,124 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-class TransferFunction(SISOLinearTimeInvariant):
-    r"""
-    A class for representing LTI (Linear, time-invariant) systems that can be strictly described
-    by ratio of polynomials in the Laplace transform complex variable. The arguments
-    are ``num``, ``den``, and ``var``, where ``num`` and ``den`` are numerator and
-    denominator polynomials of the ``TransferFunction`` respectively, and the third argument is
-    a complex variable of the Laplace transform used by these polynomials of the transfer function.
-    ``num`` and ``den`` can be either polynomials or numbers, whereas ``var``
-    has to be a :py:class:`~.Symbol`.
-
-    Explanation
-    ===========
-
-    Generally, a dynamical system representing a physical model can be described in terms of Linear
-    Ordinary Differential Equations like -
-
-            $b_{m}y^{\left(m\right)}+b_{m-1}y^{\left(m-1\right)}+\dots+b_{1}y^{\left(1\right)}+b_{0}y=
-            a_{n}x^{\left(n\right)}+a_{n-1}x^{\left(n-1\right)}+\dots+a_{1}x^{\left(1\right)}+a_{0}x$
-
-    Here, $x$ is the input signal and $y$ is the output signal and superscript on both is the order of derivative
-    (not exponent). Derivative is taken with respect to the independent variable, $t$. Also, generally $m$ is greater
-    than $n$.
-
-    It is not feasible to analyse the properties of such systems in their native form therefore, we use
-    mathematical tools like Laplace transform to get a better perspective. Taking the Laplace transform
-    of both the sides in the equation (at zero initial conditions), we get -
-
-            $\mathcal{L}[b_{m}y^{\left(m\right)}+b_{m-1}y^{\left(m-1\right)}+\dots+b_{1}y^{\left(1\right)}+b_{0}y]=
-            \mathcal{L}[a_{n}x^{\left(n\right)}+a_{n-1}x^{\left(n-1\right)}+\dots+a_{1}x^{\left(1\right)}+a_{0}x]$
+def _check_compatibility(systems):
+    """Checks compatibility between different systems"""
+    if not all(isinstance(system, LinearTimeInvariant) \
+               for system in systems):
+        return
+
+    continuous = systems[0].is_continuous
+    sampling_time = systems[0].sampling_time
+    for system in systems[1:]:
+        if system.is_continuous != continuous:
+            raise TypeError("""
+                Incompatible systems detected.
+                All systems should be either continuous-time or discrete-time.
+                Found: {} and {}""".format(systems[0], system,))
+        if system.sampling_time != sampling_time:
```

Comment:
Is this check robust for Float(), float(), Integer(), int(), Symbol, etc? I guess sampling time can be different types.